pylint/checkers/variables.py

```diff
@@ -1103,6 +1117,55 @@ def __init__(self, linter: PyLinter) -> None:
         """This is a queue, last in first out."""
         self._postponed_evaluation_enabled = False
 
+    @utils.only_required_for_messages(
+        "unbalanced-dict-unpacking",
+    )
+    def visit_for(self, node: nodes.For) -> None:
+        if not isinstance(node.target, nodes.Tuple):
+            return
+
+        targets = node.target.elts
+
+        inferred = utils.safe_infer(node.iter)
+        if inferred is None or not isinstance(inferred, DICT_TYPES):
+            return
+
+        values = self._nodes_to_unpack(inferred)
+        if values is None or len(values) == 0:
+            # no dict items returned
+            return
+
+        if isinstance(inferred, astroid.objects.DictItems):
+            # dict.items() is a bit special because values will be a tuple
+            # So as long as there are always 2 targets and values each are
+            # a tuple with two items, this will unpack correctly.
+            # Example: `for key, val in {1: 2, 3: 4}.items()`
+            if len(targets) == 2 and all(len(x.elts) == 2 for x in values):
+                return
```

Comment:
Why wouldn't ``all(len(x.elts) == len_targets for x in values)`` with ``len_target = len(targets)`` work in the general case ?

Reply:
Good question, this looks trickier than it is. The answer is: because x.elts is always 2 because dicts always unpack to a tuple of len 2. There are no dicts with anything other than (key:value). so

```
for key, val in {1: 2, 3: 4}.items():
	print(key)
```
works but this
```
for key, val, other in {1: 2, 3: 4}.items():
	print(key)
```
does not work because it unpacks to `dict_items([(1, 2), (3, 4)])`

Does that clear this up?